Sources/SwiftDocC/Model/ParametersAndReturnValidator.swift

```diff
@@ -0,0 +1,473 @@
+                variants[trait] = ParametersSection(parameters: [])
+                continue
+            }
+            
+            // Add a missing error parameter documentation if needed.
+            if trait == .objectiveC, Self.shouldAddObjectiveCErrorParameter(signatures, parameters) {
+                languageApplicableParameters["error"] = [Parameter(name: "error", contents: Self.objcErrorDescription)] // This parameter is synthesized and doesn't have a source range.
+            }
+            
+            // Ensure that the parameters are displayed in the order that they're declared.
+            var sortedParameters: [Parameter] = []
+            sortedParameters.reserveCapacity(languageApplicableParameters.count)
+            for parameter in signature.parameters {
+                if let foundParameter = languageApplicableParameters[parameter.name]?.first {
+                    sortedParameters.append(foundParameter)
+                }
+            }
+            
+            variants[trait] = ParametersSection(parameters: sortedParameters)
+        }
+        
+        // Diagnose documented parameters that's not found in any language representation's function signature.
+        for parameter in parameters where !allKnownParameterNames.contains(parameter.name) {
+            if let matchingParameterNames = argumentLabelsAndPossibleParameterNames[parameter.name] {
+                diagnosticEngine.emit(externalParameterNameProblem(parameter, knownParameterNamesForArgumentLabel: matchingParameterNames.sorted()))
+            } else {
+                diagnosticEngine.emit(extraParameterProblem(parameter, knownParameterNames: allKnownParameterNames, symbolKind: symbolKind))
+            }
+        }
+        
+        // Diagnose parameters that are documented more than once.
+        for (name, parameters) in parametersByName where allKnownParameterNames.contains(name) && parameters.count > 1 {
+            let first = parameters.first! // Each group is guaranteed to be non-empty.
+            for parameter in parameters.dropFirst() {
+                diagnosticEngine.emit(duplicateParameterProblem(parameter, previous: first, symbolKind: symbolKind))
+            }
+        }
+        
+        // Diagnose missing parameters, based on the language representation with the most parameters.
+        if let parameterOrder = signatures.values.map(\.parameters).max(by: { $0.count < $1.count })?.map(\.name) {
+            let parametersEndLocation = parameters.last?.range?.upperBound
+            
+            for parameterName in allKnownParameterNames.subtracting(["error"]) where parametersByName[parameterName] == nil {
+                // Look for the parameter that should come after the missing parameter to insert the placeholder documentation in the right location.
+                let parameterAfter = parameterOrder.drop(while: { $0 != parameterName }).dropFirst()
```

Comment:
Same here. Maybe rename `parameterAfter` to `parameterNameAfter` to make it clear this is a name, not a parameter.

Reply:
This is actually a `Parameter` value